src/sort_transform.c

```diff
@@ -59,7 +59,9 @@ transform_time_bucket(FuncExpr *func)
 	 */
 	Expr	   *second;
 
-	if (list_length(func->args) != 2 || !IsA(linitial(func->args), Const))
+	Assert(list_length(func->args) >= 2);
+
+	if (!IsA(linitial(func->args), Const) ||(list_length(func->args) == 3 && !IsA(lthird(func->args), Const)))
```

Comment:
Not sure which part you want to turn into function/macro. Line 64 checks that period and offset are constants otherwise the optimization cannot be made. I added a comment for that

Reply:
pg_ident disagrees with you about the whitespace after ||

Reply:
Agree with making this a macro. Maybe `timebucket_has_const_period_and_offset`

Reply:
ok made it a macro